src/NServiceBus.Metrics/ProbeBuilders/CriticalTimeProbeBuilder.cs

```diff
@@ -21,8 +21,11 @@ protected override void WireUp(DurationProbe probe)
             if (e.TryGetTimeSent(out timeSent))
             {
                 var endToEndTime = e.CompletedAt - timeSent;
+                string messageType;
+                e.TryGetMessageType(out messageType);
```

Comment:
If we push for C# 7 features in this project this can be `e.TryGetMessageType(out string messageType)`